mentat/code_file_manager.py

```diff
@@ -105,10 +107,15 @@ async def write_changes_to_files(
                         f"Attempted to rename file {file_edit.file_path} to existing"
                         f" file {file_edit.rename_file_path}"
                     )
-                self._add_file(file_edit.rename_file_path, code_context)
+                self._create_file(file_edit.rename_file_path, code_context)
                 self._delete_file(file_edit.file_path, code_context)
                 file_edit.file_path = file_edit.rename_file_path
 
             new_lines = file_edit.get_updated_file_lines(stored_lines)
             with open(file_edit.file_path, "w") as f:
                 f.write("\n".join(new_lines))
+
+    def get_file_checksum(self, path: Path) -> str:
```

Comment:
This could also be a method of CodeFile

Reply:
I think if you move both these methods you can remove the CodeFileManager import from code_file and then you won't have a circular import anymore.

Reply:
Same comment as above re: CodeFileManager. I was also thinking to index the embeddings database by checksum.